distribution/lib/Standard/Test/0.0.0-dev/src/Test.enso

```diff
@@ -39,9 +40,19 @@ type Test
                         State.run Spec (Spec.Value name List.Nil) <|
                             behaviors
                             State.get Spec
-                    Test_Reporter.print_report r config suite.builder
-                    new_suite = Test_Suite.Value suite.config (List.Cons r suite.specs) suite.builder
-                    State.put Test_Suite new_suite
+                    case find_parent_spec of
+                        Nothing ->
+                            # This is the root group - print report and save it to the test suite:
+                            Test_Reporter.print_report r config suite.builder
+                            new_suite = Test_Suite.Value suite.config (List.Cons r suite.specs) suite.builder
+                            State.put Test_Suite new_suite
+                        Spec.Value parent_name parent_behaviors ->
+                            # We add our nested group name to each behavior
+                            extended_behaviors = r.behaviors.map b-> b.prepend_name r.name
+
+                            # And then add the behaviors to our parent spec's list of behaviors
+                            new_spec = Spec.Value parent_name (prepend_list extended_behaviors parent_behaviors)
+                            State.put Spec new_spec
```

Comment:
IMHO the simplest and cleanest solution for handling nesting if we don't have nesting output support: we just fold the inner group into the outer group, adding the name as prefix for each test name - this way we know from where the test is coming, but the ordering of tests is preserved in the report, and all test are counted.